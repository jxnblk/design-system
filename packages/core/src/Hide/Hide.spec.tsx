import React from 'react'

import { Hide } from '..'

describe('Hide', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(<Hide />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with xs prop', () => {
    const json = rendererCreateWithTheme(<Hide xs />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with sm prop', () => {
    const json = rendererCreateWithTheme(<Hide sm />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with md prop', () => {
    const json = rendererCreateWithTheme(<Hide md />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with lg prop', () => {
    const json = rendererCreateWithTheme(<Hide lg />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with xl prop', () => {
    const json = rendererCreateWithTheme(<Hide xl />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with xxl prop', () => {
    const json = rendererCreateWithTheme(<Hide xxl />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with print prop', () => {
    const json = rendererCreateWithTheme(<Hide print />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
