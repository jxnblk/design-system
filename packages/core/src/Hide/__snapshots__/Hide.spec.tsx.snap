// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide renders 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className=""
  />
</div>
`;

exports[`Hide renders with lg prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (min-width:48em) and (max-width:63.99em) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Hide renders with md prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (min-width:40em) and (max-width:47.99em) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Hide renders with print prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media print {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Hide renders with sm prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (min-width:32em) and (max-width:39.99em) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Hide renders with xl prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (min-width:64em) and (max-width:79.99em) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Hide renders with xs prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (max-width:31.99em) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Hide renders with xxl prop 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (min-width:80em) {
  .c1 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
